logstash-core/src/main/java/org/logstash/config/ir/expression/BinaryBooleanExpression.java

```diff
@@ -49,4 +49,16 @@ public String toRubyString() {
     public String uniqueHash() {
         return Util.digest(this.getClass().getCanonicalName() + "[" + getLeft().hashSource() + "|" + getRight().hashSource() + "]");
     }
+
+    private static void ensureNotNull(final Expression expression, final String identifier,
```

Comment:
nitpick: I don't find the arbitrary newlines in this method declaration to be very readable. 

In past projects my teams have preferred one parameter per line, aligned together like the following for readability:

~~~ java
    private static void ensureNotNull(final Expression expression,
                                      final String identifier,
                                      final SourceWithMetadata meta) {
~~~

Reply:
nitpick: I'd prefer more-descriptive variable names; even though these are short methods, it's easy to get mixed up with vague names:

 - `identifier`: I'd prefer `side` or `rightOrLeft`
 - `meta`: I'd prefer `source` or `sourceWithMetadata`

Reply:
@yaauie the arbitrary new lines are kinda the style found in most of our codebase (also it's the style of ES). I think it's best to stay consistent here.

`meta` I'd keep because it's also the name used for this thing in the all our constructors upstream and downstream => better to stay consistent.
But sure, let's use `side` since it's only left or right :)